editors/code/src/main.ts

```diff
@@ -93,6 +93,12 @@ export async function activate(context: vscode.ExtensionContext) {
         activateHighlighting(ctx);
     }
     activateInlayHints(ctx);
+
+    vscode.workspace.onDidChangeConfiguration(
+        _ => ctx?.client?.sendNotification('workspace/didChangeConfiguration', { settings: "" }),
+        null,
+        ctx?.subscriptions,
+    );
```

Comment:
```suggestion
    vscode.workspace.onDidChangeConfiguration(
        _ => ctx.client.sendNotification('workspace/didChangeConfiguration', { settings: "" }),
        null,
        ctx.subscriptions,
    );
```
TypeScipt has context typing, thus if you hover over `ctx` type here, you should not see `| undefined`.

I also wonder if we could just mass the configuration here, but then we won't follow the lsp protocol, won't we?
I wish `vscode-languageclient` could automatically send the `workspace/didChangeConfiguration` for us...

Reply:
> TypeScipt has contextual typing, thus if you hover over ctx type here, you should not see | undefined.

Only the `ctx.subscriptions,` compiles, the closure expression requires `?`
<img width="772" alt="image" src="https://user-images.githubusercontent.com/2690773/77237968-2b7c6480-6bd5-11ea-863d-25b2bdd8b311.png">

> I also wonder if we could just pass the configuration here, but then we won't follow the lsp protocol, will we?

No, we should not, I've left the comment on the Rust counterpart: 
https://github.com/rust-analyzer/rust-analyzer/pull/3666/files#diff-688b34883e8efae680c3c682383af7e3R653

Reply:
Ah, right, this is a closure, and a closure can be called at any point of time, including when `ctx === undefined`.
But however, you can remove that `client?.` question mark at least, anyway.